access/slackbot/main.go

```diff
@@ -263,49 +260,71 @@ func (a *App) Run(ctx context.Context) error {
 	// Create callback server prividing a.OnSlackCallback as a callback function
 	a.callbackServer = NewCallbackServer(&a.conf, a.OnSlackCallback)
 
+	// Create separate context for Watcher in order to terminate it individually.
+	wctx, wcancel := context.WithCancel(ctx)
+
+	var jobs sync.WaitGroup
+	a.spawn = func(f func(context.Context)) {
+		jobs.Add(1)
+		go func() {
+			f(ctx)
+			jobs.Done()
+		}()
+	}
+
+	jobs.Add(1) // Start the "app" job. We have to do it for Wait() not being returned beforehand.
+	doneCh := make(chan struct{})
+	go func() {
+		jobs.Wait()
+		close(doneCh)
+	}()
```

Comment:
I'd move this inside of `shutdown` and use `jobs.Wait()` here https://github.com/gravitational/teleport-plugins/pull/53/files#diff-b0e8b9ea124c13ad8e2c86d4e5280a93R327. This way we won't cast 2 runtime-long goroutines. And code IMHO will be a bit more readable.

Reply:
I re-thinked the `Shutdown` method, now in accepts the context. I cannot do `jobs.Wait()` there.